recipes/theora/all/conanfile.py

```diff
@@ -1,182 +1,96 @@
-
-    settings = "os", "arch", "compiler", "build_type"
-    options = {
-        "shared": [True, False],
-        "fPIC": [True, False],
-    }
-    default_options = {
-        "shared": False,
-        "fPIC": True,
-    }
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+    description = "Theora is a free and open video compression format from the Xiph.org Foundation"
+    topics = "video", "video-compressor", "video-format"
 
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
 
-    @property
-    def _user_info_build(self):
-        return getattr(self, "user_info_build", self.deps_user_info)
+    # original theora.def from: "https://raw.githubusercontent.com/xiph/theora/v1.1.1/lib/theora.def"
+    # edited to change library name to just "theora"
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", self.recipe_folder, os.path.join(self.export_sources_folder,"src"))
+        copy(self, "conan-theora.def", self.recipe_folder, os.path.join(self.export_sources_folder,"src"))
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC
+            except ConanException:
+                pass
+        try:
+            del self.settings.compiler.libcxx
+        except ConanException:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except ConanException:
+            pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         self.requires("ogg/1.3.5")
 
-    def build_requirements(self):
-        if not self._is_msvc:
-            self.build_requires("gnu-config/cci.20201022")
-            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-                self.build_requires("msys2/cci.latest")
-
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version][0], strip_root=True, destination=self._source_subfolder)
-
-        source = self.conan_data["sources"][self.version][1]
-        url = source["url"]
-        filename = url[url.rfind("/") + 1:]
-        tools.download(url, filename)
-        tools.check_sha256(filename, source["sha256"])
-
-        shutil.move(filename, os.path.join(self._source_subfolder, "lib", filename))
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```

Comment:
we still keep destination 
```suggestion
        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
```